src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -6899,13 +6905,6 @@ public void ClearContent(string path)
 
             path = NormalizePath(path);
 
-            if (Directory.Exists(path))
-            {
-                string errorMsg = StringUtil.Format(SessionStateStrings.ClearDirectoryContent, path);
-                WriteError(new ErrorRecord(new NotSupportedException(errorMsg), "ClearDirectoryContent", ErrorCategory.InvalidOperation, path));
-                return;
-            }
```

Comment:
Please comment for reviewers why did we remove this?

Reply:
This wasn't removed, it was moved down below where the stream to act on is determined.  We have to figure out if there's a stream to work on (i.e., figure out if there's content that needs to be treated like a file instead of a directory) before we can decide to return without doing anything.